### PR TITLE
Fix unit conversion plugin

### PR DESCRIPTION
**Pull request by pllim**
### Description

The Unit Conversion plugin seems to be completely broken in `main` right now, at least to me. I discovered this while looking to reproduce #646 (the original use case does not even seem to apply anymore?).

In main (after you fix the `self.app._default_spectrum_viewer_reference_name` typo that is also part of this PR):

Screenshot 2022-11-11 151351

With this patch, though the slider still tries to move around in micron space and I am not sure how to fix that:

Screenshot 2022-11-11 154324

Maybe the correct fix is something completely different. 🤷 

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-2923)

### CODE REVIEW

**Comment by pllim** [comment]
@javerbukh , if you are talking about https://github.com/spacetelescope/jdaviz/pull/1398 , the two upstream PRs linked there are still in draft PRs status.

**Comment by javerbukh** [comment]
That was the work I was talking about...has there been any update from SMEs or POs on its priority?

**Comment by pllim** [comment]
No idea... Maybe @camipacifici or @orifox know.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def vue_unit_conversion(self, *args, **kwargs):
                 self.app.add_data(converted_spec, new_label)
 
         self.app.add_data_to_viewer(
-            self.app._default_spectrum_viewer_reference_name,
+            self._default_spectrum_viewer_reference_name,
```
Comment (bmorris3) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:165]: Sorry about leaving this straggler in there! 

There's one more to catch here: https://github.com/spacetelescope/jdaviz/pull/1839/files#diff-6fd13e1a6ca69516fb610f228b7e5a0f521ea2fbba7682350c3b04c59c8434b7R143

Comment (pllim) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:165]: Done. Thanks!

Comment (pllim) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:165]: This is now part of #1970

**Comment by pllim** [comment]
I am closing this without merge since refactoring is going to happen anyway. The uncontroversial bug fix is now part of #1970.

**Pull request closed without merging**